File: app/src/main/java/com/wjf/self_demo/activity/MainActivity.java
package com.wjf.self_demo.activity;

import com.wjf.self_demo.R;
import com.wjf.self_demo.databinding.ActivityMainBinding;
import com.wjf.self_library.common.BaseActivity;

/**
 * @author Wangjf2-DESKTOP
 */
public class MainActivity extends BaseActivity<ActivityMainBinding> {

    @Override
    public int setLayout() {
        return R.layout.activity_main;
    }

    @Override
    protected void initView() {
/*        view.recyclerViewune.setAdapter(
                new ListAdapter(
                        this,
                        Arrays.asList(
                                new UserInfo("王佳峰", "大帅哥"),
                                new UserInfo("章志民", "大帅哥"),
                                new UserInfo("吴一鸣", "大帅哥"))));
        view.recyclerView.setLayoutManager(
                new LinearLayoutManager(this, RecyclerView.VERTICAL, false));*/
    }

    @Override
    protected void initData() {
    }
}

File: app/src/main/java/com/wjf/self_demo/activity/MainActivity.kt
package com.wjf.self_demo.activity

import com.orhanobut.logger.Logger
import com.wjf.self_demo.databinding.ActivityMainBinding
import org.jxxy.debug.corekit.common.BaseActivity

/**
 * @author Wangjf2-DESKTOP
 */
class MainActivity :
    BaseActivity<ActivityMainBinding>() {

    override fun initView() {
        /*        view.recyclerViewune.setAdapter(
                new ListAdapter(
                        this,
                        Arrays.asList(
                                new UserInfo("王佳峰", "大帅哥"),
                                new UserInfo("章志民", "大帅哥"),
                                new UserInfo("吴一鸣", "大帅哥"))));
        view.recyclerView.setLayoutManager(
                new LinearLayoutManager(this, RecyclerView.VERTICAL, false));*/

        val sp = getSharedPreferences(packageName, MODE_PRIVATE)
        val edit = sp.edit()
        edit.putString("key", "")
        edit.commit()
        val value = sp.getString("key", "sss")
        Logger.d("key is ${value ?: "aaa"}")
        Logger.d(
            "key is ${
            value?.ifEmpty {
                "aaaaa"
            }
            }"
        )
    }

    override fun bindLayout(): ActivityMainBinding {
        return ActivityMainBinding.inflate(layoutInflater)
    }

    override fun subscribeUi() {
    }
}
